lib/packages/docutils/rst.nim

```diff
@@ -1018,15 +1036,43 @@ proc fixupEmbeddedRef(n, a, b: PRstNode) =
   for i in countup(0, sep - incr): a.add(n.sons[i])
   for i in countup(sep + 1, n.len - 2): b.add(n.sons[i])
 
-proc whichRole(sym: string): RstNodeKind =
-  case sym
+const supportedLanguages = ["nim", "yaml", "python", "java", "c",
```

Comment:
is that supposed to stay in sync with 
```nim
const
  sourceLanguageToStr*: array[SourceLanguage, string] = ["none",
    "Nim", "C++", "C#", "C", "Java", "Yaml", "Python"]
```
? if so, refactor (DRY)

Reply:
ideally, yes, but mirror should be maintained anyway, because `c#` role would not be allowed according to RST spec:

> A role name is a single word consisting of alphanumerics plus isolated internal hyphens, underscores, plus signs, colons, and periods; no whitespace or other characters are allowed.